File: AudioDelay/AudioRecorder/AudioRecorder.cs
using AudioDelay.Args;

namespace AudioDelay.AudioRecorder;

public abstract class AudioRecorder: IDisposable
{
  protected AudioRecorder(Arguments args) {}

  public abstract void Play();
  public abstract void StopPlayback();
  public abstract void Start();
  public abstract void Stop();

  public abstract void Dispose();
}

File: src/java/src/AudioRecorder/AudioRecorder.java
package AudioRecorder;

import Args.Arguments;

import javax.sound.sampled.AudioInputStream;

public abstract class AudioRecorder
{
    protected Arguments _args;
    protected AudioRecorder(Arguments args) {
        _args = args;
    }
    public abstract void Play();
    public abstract void StopPlayback();
    public abstract void Start();
    public abstract void Stop();
}
